src/shader_recompiler/backend/spirv/emit_spirv_quad_rect.cpp

```diff
@@ -0,0 +1,309 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <sirit/sirit.h>
+#include "shader_recompiler/backend/spirv/emit_spirv_quad_rect.h"
+
+namespace Shader::Backend::SPIRV {
+
+using Sirit::Id;
+
+struct QuadRectListEmitter : public Sirit::Module {
+    explicit QuadRectListEmitter(size_t num_attribs_)
```

Comment:
Should pass supported SPIR-V version to `Sirit::Module` like so: https://github.com/shadps4-emu/shadPS4/blob/c2e9c877dd82e9bf79c8ede7bc8cffb591a31c76/src/shader_recompiler/backend/spirv/spirv_emit_context.cpp#L68

Otherwise emits a validation error on Vulkan 1.2.

Reply:
Can also pass a fixed, lower version (<= 1.5 I think) as I suppose newer isn't needed.